massa-execution-worker/src/execution.rs

```diff
@@ -1079,6 +1088,31 @@ impl ExecutionState {
         // apply execution output to final state
         self.apply_final_execution_output(exec_out);
         debug!("execute_final_slot: execution finished & result applied");
+
+        // update versioning statistics
+        if let Some((block_id, storage)) = exec_target {
+            if let Some(_block) = storage.read_blocks().get(block_id) {
+                let slot_ts_ = get_block_slot_timestamp(
+                    self.config.thread_count,
+                    self.config.t0,
+                    self.config.genesis_timestamp,
+                    *slot,
+                );
+
+                if let Ok(slot_ts) = slot_ts_ {
+                    // TODO - Next PR: use block header network versions - default to 0 for now
+                    self.mip_store
+                        .update_network_version_stats(slot_ts, Some((0, 0)));
+                } else {
+                    warn!("Unable to get slot timestamp for slot: {} in order to update mip_store stats", slot);
+                }
+            } else {
+                warn!(
+                    "Could not find block id: {} in storage in order to update mip_store stats",
+                    block_id
+                );
+            }
+        }
```

Comment:
As it's not the main goal of `execute_final_slot`, I think it makes sense to put this in it's own method `update_versioning_stats(exec_target)`.

Reply:
done